Code/Tools/AssetProcessor/native/AssetManager/AssetRequestHandler.cpp

```diff
@@ -107,6 +107,46 @@ namespace
                     auto resultMove = relocationInterface->Move(
                         messageData.m_message->m_fromPath, messageData.m_message->m_toPath, false, true, true, true);
 
+                    if (resultMove.IsSuccess())
+                    {
+                        AssetProcessor::RelocationSuccess success = resultMove.TakeValue();
+
+                        // The report can be too long for the AZ_Printf buffer, so split it into individual lines
+                        AZStd::string report =
+                            relocationInterface->BuildChangeReport(success.m_relocationContainer, success.m_updateTasks);
+                        AzFramework::StringFunc::Tokenize(report.c_str(), lines, "\n");
+
+                        for (const AZStd::string& line : lines)
+                        {
+                            AZ_Printf(AssetProcessor::ConsoleChannel, (line + "\n").c_str());
+                        }
+                    }
+                    break;
+                }
+                case AssetChangeReportRequest::ChangeType::CheckDelete:
+                {
+                    auto resultCheck = relocationInterface->Delete(messageData.m_message->m_fromPath);
+
+                    if (resultCheck.IsSuccess())
+                    {
+                        AssetProcessor::RelocationSuccess success = resultCheck.TakeValue();
+
+                        // The report can be too long for the AZ_Printf buffer, so split it into individual lines
+                        AZStd::string report =
+                            relocationInterface->BuildChangeReport(success.m_relocationContainer, success.m_updateTasks);
+                        AzFramework::StringFunc::Tokenize(report.c_str(), lines, "\n");
+
+                        for (const AZStd::string& line : lines)
+                        {
+                            AZ_Printf(AssetProcessor::ConsoleChannel, (line + "\n").c_str());
```

Comment:
repeat: `AZ::Debug::Trace::Output` should be used here to avoid format string injection as well as to avoid truncation.
`AZ_Printf` uses `AZ::Debug::Trace::Printf` formats the output into a 2048 character buffer, while `AZ::Debug::Trace::Output` just outputs a c-string without any formatting and truncation.

```suggestion
                            AZ::Debug::Trace::Instance().Output(AssetProcessor::ConsoleChannel, (line + "\n").c_str());
```

Reply:
Done.